File: leet_2373_largest_local.go
func largestLocal(g [][]int) [][]int {
	R := len(g) - 2
	C := len(g[0]) - 2

	res := make([][]int, R)
	for i := range res {
		res[i] = make([]int, C)
	}

	r := -1
	for r < R - 1 {
        r++
		c := -1
		for c < C - 1 {
            c++
			res[r][c] = Max_Local(g, r, c)
		}
	}

	return res
}

// helper

func Max_Local(g [][]int, r int, c int) int {
	res := -1
	rr := r - 1
	cc := 0
	for rr < r + 2 {
		rr++
        cc = c - 1
		for cc < c + 2 {
            cc++
			val := g[rr][cc]
			if val > res {
				res = val
			}
		}
	}
	return res
}

File: leet_2373_largest_local.ts
function largestLocal(g: number[][]): number[][] {
    let R: number = g.length - 2
    let C: number = g[0].length - 2

    //  Does not work

    // let res: number[][] = new Array(R).fill(Array(C, 0))
    // let res: number[][] = new Array(R).fill(Array(C).fill(0))


    //  Correct way to fill a 2-D array

    // let res: number[][] = Array.from({ length: R }, () => Array(C).fill(0))
    let res: number[][] = Array(R).fill(0).map(() => Array(C).fill(0));

    let r: number = -1, c: number
    while (++r < R) {
        c = -1
        while (++c < C) {
            res[r][c] = Max_Local(g, r, c)
        }
    }
    return res
};

// helper

function Max_Local(g: number[][], r: number, c: number) {
    let res: number = -1
    let rr: number = r - 1
    let cc: number
    while (++rr < r + 3) {
        cc = c - 1
        while (++cc < c + 3) {
            let val = g[rr][cc]
            res = res > val ? res : val
        }
    }
    return res
}
